src/os/bluestore/BlueStore.cc

```diff
@@ -1823,21 +1822,22 @@ void BlueStore::ExtentMap::update(KeyValueDB::Transaction t,
 	  if (len > cct->_conf->bluestore_extent_map_shard_max_size) {
 	    // we are big; reshard ourselves
 	    request_reshard(p->offset, endoff);
-	    return;
 	  }
 	  // avoid resharding the trailing shard, even if it is small
-	  if (n != shards.end() &&
-	      len < g_conf->bluestore_extent_map_shard_min_size) {
+	  else if (n != shards.end() &&
+		   len < g_conf->bluestore_extent_map_shard_min_size) {
 	    // we are small; combine with a neighbor
 	    if (p == shards.begin() && endoff == OBJECT_MAX_SIZE) {
 	      // we are an only shard
 	      request_reshard(0, OBJECT_MAX_SIZE);
+	      return;
```

Comment:
We'll exist the loop at this iteration and return the function anyway hence just a noice